storage/remote/write_handler.go

```diff
@@ -453,26 +522,26 @@ func (h *writeHandler) appendV2(app storage.Appender, req *writev2.Request, rs *
 			e := ep.ToExemplar(&b, req.Symbols)
 			ref, err = app.AppendExemplar(ref, ls, e)
 			if err == nil {
-				rs.exemplars++
+				rs.Exemplars++
 				continue
 			}
 			// Handle append error.
-			// TODO(bwplotka): I left the logic as in v1, but we might want to make it consistent with samples and histograms.
-			// Since exemplar storage is still experimental, we don't fail in anyway, the request on ingestion errors.
 			if errors.Is(err, storage.ErrOutOfOrderExemplar) {
-				outOfOrderExemplarErrs++
-				level.Debug(h.logger).Log("msg", "Out of order exemplar", "err", err.Error(), "series", ls.String(), "exemplar", fmt.Sprintf("%+v", e))
+				outOfOrderExemplarErrs++ // Maintain old metrics, but technically not needed, given we fail here.
+				level.Error(h.logger).Log("msg", "Out of order exemplar", "err", err.Error(), "series", ls.String(), "exemplar", fmt.Sprintf("%+v", e))
+				badRequestErrs = append(badRequestErrs, fmt.Errorf("%w for series %v", err, ls.String()))
 				continue
 			}
-			level.Debug(h.logger).Log("msg", "Error while adding exemplar in AppendExemplar", "series", ls.String(), "exemplar", fmt.Sprintf("%+v", e), "err", err)
+			// TODO(bwplotka): Are we confident enough to 500 on exemplars in Prometheus?
```

Comment:
I don't think the exemplar storage has really been touched since I first wrote it, so I would lean towards no, but I could be convinced otherwise by anyone who has worked on the exemplar storage since then.

Reply:
Do we see a lot of internal errors? We can take easier path and keep the internal errors silent for now 💪🏽

Reply:
I actually don't know tbh, I don't use them much myself, so I'd defer to anyone who more experience with them. I can ask our internal platform team or Mimir next week.

Reply:
Got some context here from @bboreham. Mimir recently moved from rejecting data that contained invalid exemplars to only rejecting the invalid exemplars themselves [here](https://github.com/grafana/mimir/pull/8224). I think we should default to doing the same, and if there's an ask for it in the future we could add a `strict` option and consider changing the default.

Reply:
Hm, not quite, sounds like Mimir still fails on some exemplar 400 failures: https://github.com/grafana/mimir/pull/8224/files#r1684633427

Nevertheless 400 fail never stops sample ingestion, so we are good here? I'm more concerned about 500 🤔

Reply:
Addressed as discussed on Slack - keep 400 but make 500 silent for callers for now with todo for strict mode if requested by users